consensus/src/round_manager.rs

```diff
@@ -378,18 +347,42 @@ impl RoundManager {
                 })?;
             self.block_store
                 .add_certs(&sync_info, self.create_block_retriever(author))
-                .await
-                .map_err(|e| {
-                    warn!("Fail to sync up to {}: {:?}", sync_info, e);
-                    e
-                })?;
+                .await?;
 
             // Update safety rules and round_state and potentially start a new round.
             self.process_certificates().await?;
         }
         Ok(())
     }
 
+    /// The function makes sure that it ensures the message_round equal to what we have locally,
+    /// brings the missing dependencies from the QC and LedgerInfo of the given sync info and
+    /// update the round_state with the certificates if succeed.
+    /// Returns Error in case sync mgr failed to bring the missing dependencies.
+    /// We'll try to help the remote if the SyncInfo lags behind and the flag is set.
+    pub async fn ensure_round(
```

Comment:
ensure_round_and_sync_up
verbose but correct..

Reply:
fine, I hate you

Reply:
I like @davidiw.

Reply:
> fine, I hate you

💩